### PR TITLE
Add/editor seo preview

### PR DESCRIPTION
**Pull request by dmsnell**
Building the preview pane in the editor preview window.

Test live: https://calypso.live/?branch=add/editor-seo-preview

### CODE REVIEW

**Comment by roundhill** [comment]
Looking great @dmsnell! I'm wondering what we should do for small screens, though... we'd have to maybe change the tabs on the left to a drop down at the top?

**Comment by dmsnell** [comment]
Not sure @dan. I can try something like that. Do you happen to have an example where we conditionally render based on viewport width? (JS measuring width instead of CSS adjusting style)

cc: @mtias

**Comment by roundhill** [comment]
@dmsnell yeah, check out `section-nav`: https://github.com/Automattic/wp-calypso/tree/master/client/components/section-nav. Might be a bit too complex for this. We could also maybe stack all of the sections vertically as well, and show the preview in each one?

p.s. I'm @roundhill here ;)

File: client/components/seo-preview-pane/index.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+const ComingSoonMessage = translate => (
+	<div className="seo-preview-pane__message">
+		{ translate( 'Coming Soon!' ) }
+	</div>
+);
+
+const GooglePreview = site =>
+	<SearchPreview
+		title={ site.name }
```
Comment (roundhill) [inline client/components/seo-preview-pane/index.jsx]: Since we're in the editor, this should be a post preview instead of a site preview.

File: client/components/seo-preview-pane/style.scss [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	background-color: $white;
+	border-right: 1px solid lighten( $gray, 20% );
+}
+
+.seo-preview-pane__explanation {
+	display: flex;
+	flex-direction: column;
+	margin-top: 40px;
+	margin-bottom: 40px;
+	margin-left: 20px;
```
Comment (roundhill) [inline client/components/seo-preview-pane/style.scss]: You can shorten this to: `margin: 40px 0 40px 20px`

File: client/components/seo-preview-pane/index.jsx (hunk 1 above)
Comment (dmsnell) [inline client/components/seo-preview-pane/index.jsx]: Thanks @roundhill - it's not a finished product, but this PR and PRs built on top of it are getting large and hard to manage. I was hoping to push this out as a stepping-stone to the final feature.

Comment (roundhill) [inline client/components/seo-preview-pane/index.jsx]: > I was hoping to push this out as a stepping-stone to the final feature.

Ah, ok. LGTM then :) :shipit:

File: client/components/seo-preview-pane/style.scss (hunk 2 above)
Comment (dmsnell) [inline client/components/seo-preview-pane/style.scss]: Updated and squashed commit - trying to keep the commit log small

**Comment by dmsnell** [comment]
> check out section-nav

@roundhill it looks like we're just checking `window.innerWidth` inside of the JS there. I have created #7098 to address this.

**Comment by folletto** [comment]
screen shot 2016-07-29 at 16 24 52


This merge seems breaking the WebPreview everywhere else (i.e. Themes → ••• → Live Demo).

We need to make sure the `gridicon-share` icon shows up exclusively on the Editor.

---

Also please remember to always add visual record on PRs (before/after or just after if before doesn't apply).
